android armv7 compile error with LTO

clang android LTO bug (only armv7 affected)
https://stackoverflow.com/questions/56238019/linker-error-when-enabling-link-time-optimization-in-ndk
Tested with android ndk v20.0.5594570 (android api 29)

```
[100%] Linking C shared library ../../lib/libddsc.so
/data/jenkins/Android/Sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld.gold: fatal error: Optimization level must be between 0 and 3
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/core/CMakeFiles/ddsc.dir/build.make:2401: lib/libddsc.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:288: src/core/CMakeFiles/ddsc.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```
It is caused by LTO and a bug with cmake <-> clang.
The archs x86, x86_64 and armv8 are not affected and work as expected with LTO,

I already fixed it and open a pull request for this.

--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -39,14 +39,12 @@ if(ENABLE_LTO)
   check_ipo_supported(RESULT ipo_supported OUTPUT error)
   if(ipo_supported)
     if(ANDROID AND "${CMAKE_ANDROID_ARCH_ABI}" STREQUAL "armeabi-v7a")
-      if("${CMAKE_ANDROID_ARCH_ABI}" STREQUAL "armeabi-v7a")
-        # clang android LTO bug (only armv7 affected)
-        # https://stackoverflow.com/questions/56238019/linker-error-when-enabling-link-time-optimization-in-ndk
-        # Fixed in newer clang versions: https://reviews.llvm.org/D79919
-        # But fixed here to keep supported android api level as low as possible
-        # Tested with android ndk v20.0.5594570 (android api 29)
-        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -flto=full")
-      endif()
+      # clang android LTO bug (only armv7 affected)
+      # https://stackoverflow.com/questions/56238019/linker-error-when-enabling-link-time-optimization-in-ndk
+      # Fixed in newer clang versions: https://reviews.llvm.org/D79919
+      # But fixed here to keep supported android api level as low as possible
+      # Tested with android ndk v20.0.5594570 (android api 29)
+      set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -flto=full")
     else()
       set_property(TARGET ddsc PROPERTY INTERPROCEDURAL_OPTIMIZATION_RELEASE TRUE)
       set_property(TARGET ddsc PROPERTY INTERPROCEDURAL_OPTIMIZATION_MINSIZEREL TRUE)

--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -38,10 +38,21 @@ if(ENABLE_LTO)
   include(CheckIPOSupported)
   check_ipo_supported(RESULT ipo_supported OUTPUT error)
   if(ipo_supported)
-    set_property(TARGET ddsc PROPERTY INTERPROCEDURAL_OPTIMIZATION_RELEASE TRUE)
-    set_property(TARGET ddsc PROPERTY INTERPROCEDURAL_OPTIMIZATION_MINSIZEREL TRUE)
-    #-flto and debugging info do not seem to go well together
-    #set_property(TARGET ddsc PROPERTY INTERPROCEDURAL_OPTIMIZATION_RELWITHDEBINFO TRUE)
+    if(ANDROID AND "${CMAKE_ANDROID_ARCH_ABI}" STREQUAL "armeabi-v7a")
+      if("${CMAKE_ANDROID_ARCH_ABI}" STREQUAL "armeabi-v7a")
+        # clang android LTO bug (only armv7 affected)
+        # https://stackoverflow.com/questions/56238019/linker-error-when-enabling-link-time-optimization-in-ndk
+        # Fixed in newer clang versions: https://reviews.llvm.org/D79919
+        # But fixed here to keep supported android api level as low as possible
+        # Tested with android ndk v20.0.5594570 (android api 29)
+        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -flto=full")
+      endif()
+    else()
+      set_property(TARGET ddsc PROPERTY INTERPROCEDURAL_OPTIMIZATION_RELEASE TRUE)
+      set_property(TARGET ddsc PROPERTY INTERPROCEDURAL_OPTIMIZATION_MINSIZEREL TRUE)
+      #-flto and debugging info do not seem to go well together
+      #set_property(TARGET ddsc PROPERTY INTERPROCEDURAL_OPTIMIZATION_RELWITHDEBINFO TRUE)
+    endif()
   endif()
 endif()
 

